active_learning_single_annotator/active_learning_single_annotator.ipynb

```diff
@@ -0,0 +1,639 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dataset = ImageClassificationDataset.from_folder('./256_ObjectCategories/')\n",
+    "dataset = dataset.replace(257, 256) # no class class in dataset is labeled as 257, we need to reindex\n",
+    "\n",
+    "# Split data into train and test\n",
+    "df_train, df_test = train_test_split(dataset, test_size=0.33, random_state=123)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The train data will further be split into a labeled and unlabeled part, `df_labeled` and `df_unlabeled` respectively. \n",
+    "\n",
+    "We will use the labels in `df_labeled` to train the model while `df_unlabeled` will simulate active learning by allowing us to artificially insert more labeled data in between rounds.\n",
+    "\n",
+    "We will arbitrarily choose to start with `num_labeled_per_class = 8`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_labeled(dataset,  num_labeled_per_class=8):\n",
+    "    \"\"\"Splits provided dataset into two datasets. With df_labeled containing num_labeled_per_class labeles for \n",
+    "    each class and df_unlabeled containing the rest of the rows in dataset\"\"\"\n",
+    "    \n",
+    "    df_labeled = dataset.groupby(\"label\").sample(n=num_labeled_per_class, random_state=123)\n",
+    "    labeled_index = list(df_labeled.index)\n",
+    "    unlabeled_index = [i for i in range(len(dataset)) if i not in labeled_index]\n",
+    "    df_unlabeled = dataset.iloc[unlabeled_index]\n",
+    "    df_unlabeled = df_unlabeled.reset_index(drop=True)\n",
+    "    df_labeled = df_labeled.reset_index(drop=True)    \n",
+    "    return df_labeled, df_unlabeled\n",
+    "\n",
+    "# Split the train data into labeled and unlabeled with 8 labeled per each class\n",
+    "df_labeled, df_unlabeled = get_labeled(df_train, num_labeled_per_class=8)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Train model on labeled data & obtain predicted probabilites for unlabeled data\n",
+    "\n",
+    "First, we train our model on the labeled data obtained from `get_labeled` and get the probabilities for the unlabeled data. The train function returns our `predictor` fitted to `df_labeled` and `pred_probs_unlabeled` which are the predicted probabilities for examples that do not have any annotator labels (they correspond directly with the rows in `df_unlabeled`). These predicted probabilities will later be used to compute the active learning score.\n",
+    "\n",
```

Comment:
Lots of feedback on the above sentence and code structure:

Reply:
say: "predicted class probabilities" instead of "predicted probabilities" which are more vague

Reply:
we do not fit a model to the pred_probs_unlabeled, we run inference using an already-trained model to produce the pred_probs_unlabeled

Reply:
split your helper methods up into two functions:

`train()` and `predict_proba()` so it looks like this:
```
predictor_model = train(df_labeled)
pred_probs_unlabeled = predict_proba(df_unlabeled , predictor_model)
```

or even better would be if the second line could just be:
```
pred_probs_unlabeled = predictor_model.predict_proba(df_unlabeled)
```

Reply:
this will make all the discussion more clear

Reply:
Remember we are not in multiple label setting here. So instead of: "examples that do not have any annotator labels" 

Say:  "examples from the pool of unlabeled data"